admin_tests/preprints/test_views.py

```diff
@@ -183,8 +183,8 @@ def test_correct_view_permissions(self, user, preprint, plain_view):
         request = RequestFactory().get(reverse('preprints:preprint', kwargs={'guid': preprint._id}))
         request.user = user
 
-        response = plain_view.as_view()(request, guid=preprint._id)
-        assert response.status_code == 302
+        with pytest.raises(PermissionDenied):
+            plain_view.as_view()(request, guid=preprint._id)
```

Comment:
hmm, this looks like something we need to fix on the permission side instead of the test itself

Reply:
might be caused by DRF, where permission check now require all permissions instead of any

>> permission_required = ('osf.view_preprint',)
>> user.has_perms(permission_required)
>> False
>> permission_required = ('osf.view_preprint', 'osf.change_preprint',)
>> user.has_perms(permission_required)
>> True

This is why `test_change_preprint_provider` passes while `test_correct_view_permissions` fail

Reply:
Final thoughts: I agree with @John Tordoff  that this is more on the test than the code. The behavior of `user.check_perms` is the same between [1.11](https://docs.djangoproject.com/en/1.11/ref/contrib/auth/#django.contrib.auth.models.User.has_perms) and [3.2](https://docs.djangoproject.com/en/3.2/ref/contrib/auth/#django.contrib.auth.models.User.has_perms). In addition, the redirection behavior when permission is denied depends on if `user.is_authenticated` is True. With it [always being set to `True`](https://github.com/CenterForOpenScience/osf.io/blob/d07206d3a7b4a26e440c66131531627170e9b1c9/osf/models/user.py#L621-L623), `PermissionDenied` should be raised when there no permission or when permission is incorrect. One of the failed test should be renamed from `test_correct_view_permissions` to `test_incorrect_view_permissions` to avoid confusion.